qa/rpc-tests/sc_mempool_conflict.py

```diff
@@ -0,0 +1,190 @@
+#!/usr/bin/env python3
+# Copyright (c) 2014 The Bitcoin Core developers
+# Copyright (c) 2018 The Zencash developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.test_framework import ForkHeights
+from test_framework.authproxy import JSONRPCException
+from test_framework.util import initialize_chain_clean, get_epoch_data, swap_bytes, \
+    start_nodes, connect_nodes_bi, mark_logs
+from test_framework.mc_test.mc_test import *
+from decimal import Decimal
+
+DEBUG_MODE    = 1
+NUMB_OF_NODES = 2
+EPOCH_LENGTH  = 0
+
+FT_SC_FEE      = Decimal('0')
+MBTR_SC_FEE    = Decimal('0')
+CERT_FEE       = Decimal('0.00015')
+PARAMS_NAME = "sc"
+
+# This script reproduces an issue introduced with non-ceasable sidechains, where conflicting
+# certificates and transactions may be submitted in the mempool, creating undefined behaviors.
+# Briefly, the test executes the following steps:
+# - create a non-ceasable sidechain
+# - submit a certificate and include that in a block
+# - submit a transaction explicitly spending one of the certificate output; do not include
+#   this transaction in a block
+# - Invalidate the block containing the certificate; the certificate gets pushed back to the
+#   mempool, creating a conflict with the transaction
+# - Under normal circumstances, the transaction should be purged from the mempool
```

Comment:
Do you think it could add value here to include a reference to the check that is performed by the mempool on Zend? 
```suggestion
# - Under normal circumstances, the transaction should be purged from the mempool (see `CTxMemPool::checkTxImmatureExpenditures()` inside `txmempool.cpp`)
```

Reply:
Thanks, I added the reference in the comment, as suggested